### PR TITLE
add LCOV export support, generalized convert command

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### coverlib: add support for exporting LCOV and merging profiles

Plus some other improvements:
 - determine format from file extension
 - generalized Import/Export functions
 - improved file renaming implementation

#### add convert command

This commit adds a generalized converter which can import profiles
from multiple files (Go cover or LCOV format), merge the profiles, and
export to either LCOV or Codecover JSON format.

This generalized tool supersedes the existing tools
(gocover2json, lcov2json).

- - -
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/code-cov-utils/3)

### CODE REVIEW

**Comment by RaduBerinde** [comment]
@srosenberg Updated to sum hit counts when merging profiles, as discussed offline.

File: coverlib/profiles.go [hunk 1]
```diff
@@ -68,3 +71,15 @@ func (p *Profiles) String() string {
 	}
 	return buf.String()
 }
+
+// MergeWith merges in the given profiles.
+//
+// For any given line, the resulting hit count is the maximum between the hit
```
Comment (srosenberg) [inline coverlib/profiles.go]: Comment is stale now that `MergeWith` computes the sum of the hit counts.

**Review by srosenberg — APPROVED** [review]
Nice!

File: coverlib/profiles.go (hunk 1 above)
Comment (RaduBerinde) [reply coverlib/profiles.go]: Good catch, fixed. TFTR!

**Pull request merged**